std/experimental/allocator/building_blocks/kernighan_ritchie.d

```diff
@@ -525,6 +530,15 @@ struct KRRegion(ParentAllocator = NullAllocator)
         return null;
     }
 
+    private void ensureCorrectSize()
+    {
+        import std.math : abs;
+
+        if (root && (abs(payload.length - root.size) < Node.sizeof)) {
```

Comment:
Also to fix gaps at the end of the memory, I modified ```coalesce``` to take a memory end parameter to find any possible gaps between ```node.start + length``` and ```memoryEnd```.